File: apple/Sources/ReactNativeWatchOS/Modifiers/ScrollContentBackground.swift
import SwiftUI

/// SwiftUI `.scrollContentBackground(_:)`. Controls whether the scroll
/// view's background is visible (e.g. hide a `List`/`Form` background to
/// reveal a custom one underneath). Available watchOS 9+.
enum RNWScrollContentBackgroundModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("scrollContentBackground") { view, params, _ in
            AnyView(view.scrollContentBackground(
                RNWVisibilityParser.parse(params.string("visibility"))
            ))
        }
    }
}

File: src/modifiers/scrollContentBackground.ts
import { createModifier } from './createModifier';

/// Visibility maps to SwiftUI's `Visibility` enum.
export type ScrollContentBackgroundVisibility =
  | 'automatic'
  | 'visible'
  | 'hidden';

export interface ScrollContentBackgroundParams {
  visibility: ScrollContentBackgroundVisibility;
}

/// SwiftUI `.scrollContentBackground(_:)`. Controls whether the scroll
/// view's background is visible (e.g. hide a `List`/`Form` background to
/// show a custom one underneath). watchOS 9+.
export function scrollContentBackground(
  visibility: ScrollContentBackgroundVisibility
): ReturnType<typeof createModifier>;
export function scrollContentBackground(
  params: ScrollContentBackgroundParams
): ReturnType<typeof createModifier>;
export function scrollContentBackground(
  a: ScrollContentBackgroundVisibility | ScrollContentBackgroundParams
) {
  if (typeof a === 'string') {
    return createModifier('scrollContentBackground', { visibility: a });
  }
  return createModifier('scrollContentBackground', a);
}
